keras/engine/training.py

```diff
@@ -1831,8 +1831,12 @@ def generate_arrays_from_file(path):
                                  str(validation_data))
             val_x, val_y, val_sample_weights = self._standardize_user_data(
                 val_x, val_y, val_sample_weight)
+            val_data = val_x + val_y + val_sample_weights
+            if self.uses_learning_phase and not isinstance(K.learning_phase(),
+                                                           int):
```

Comment:
I mean these checks: https://github.com/fchollet/keras/blob/master/keras/engine/training.py#L1447 and https://github.com/fchollet/keras/blob/master/keras/engine/training.py#L1462